docs/current_docs/guides/snippets/build-test-publish-java-spring/go/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"context"
+)
+
+type MyModule struct{}
+
+func (m *MyModule) Build(ctx context.Context, source *Directory) (string, error) {
+
+	mariadb := dag.Container().
+		From("mariadb:10.11.2").
+		WithEnvVariable("MARIADB_USER", "petclinic").
+		WithEnvVariable("MARIADB_PASSWORD", "petclinic").
+		WithEnvVariable("MARIADB_DATABASE", "petclinic").
+		WithEnvVariable("MARIADB_ROOT_PASSWORD", "root").
+		WithExposedPort(3306).
+		AsService()
+
+	app := dag.Java().
+		WithJdk("17").
+		WithMaven("3.9.5").
+		WithProject(source.WithoutDirectory("dagger")).
+		Maven([]string{"-X", "-Dspring.profiles.active=mysql", "clean", "package"})
```

Comment:
The issue seems to be that you need the database bound and configured (i.e. the `build :=` part below) _before_ running this command. I'm not seeing a way to do that with the Java module though (cc @jcsirot). Maybe skip the module for now?